service.py

```diff
@@ -134,15 +138,12 @@ def get_gfs_from_biospecimen(self, biospecimen_id):
         Returns the links of biospecimen
         """
         resp = requests.get(
-            self.api+'/biospecimens/'+biospecimen_id)
-        if resp.status_code != 200 and len(resp.json()['results']) != 1:
-            return 'Biospecimen does not exist'
-        row = resp.json()
-        resp = requests.get(
-            self.api +
-            row['_links']['biospecimen_genomic_files']+'&limit=100')
+            self.api+'/genomic-files?biospecimen_id='+biospecimen_id +
+            '&limit=100')
         if resp.status_code != 200 and len(resp.json()['results']) <= 0:
-            return 'No associated biospecimen-genomic-files found'
+            raise DataserviceException(
+                f'No associated genomic-files found for '
```

Comment:
The length can't be < 0.
If the server response with something other than status 200 here, then it's for an entirely different reason than files not being found. It should sleep a few seconds and then repeat the request at least once if a non-200 status is returned.

Reply:
one of the issue of lambda is it times out before the status gets green

Reply:
So what happens to the failed one? It just stays failed? I think that's bad.

Reply:
i see your point. currently it just skips the row. when i am running locally i am checking for non 200 and recalling it again. but i cannot do the same in lambda because it wastes resources and that's how amazon makes money.

Reply:
How much money are we talking about?